packages/elements/src/clock/__test__/clock.interactive.test.js

```diff
@@ -132,11 +144,78 @@ describe('clock/Interactive', () => {
 
       await onTapstart(minutesSegment, el);
-      setTimeout(() => createKeyboardEvent(minutesSegment, InputKey.arrowUp));
+      setTimeout(() => createKeyboardEvent(el, InputKey.ArrowUp));
       const offsetEvent = await oneEvent(el, 'offset-changed');
 
       expect(offsetChangedCount, 'offset-changed count should be 1').to.be.equal(1);
       expect(offsetEvent.detail.value, '  should be 60').to.be.equal(60);
     });
+    
+    describe('Accessibility', () => {
+      it('Should have role="spinbutton" and focusable', async () => {
+        expect(el.getAttribute('role')).to.be.equal('spinbutton');
+        expect(el.getAttribute('tabindex')).to.be.equal('0');
+      });
+      it('Should increase hour value and update aria-valuetext when Arrow Up is pressed on hour segment', async () => {
+        await onTapstart(hoursSegment, el);
+        createKeyboardEvent(el, InputKey.ArrowUp);
+        await elementUpdated(el);
+        await nextFrame();
+  
+        expect(el.getAttribute('aria-valuetext')).to.be.equal('Time: 01:00');
+        expect(el.getAttribute('aria-valuenow')).to.be.equal(el.displayTime.toString());
+      });
+      it('Should decrease hour value and update aria-valuetext when Arrow Down is pressed on hour segment', async () => {
+        await onTapstart(hoursSegment, el);
+        createKeyboardEvent(el, InputKey.ArrowDown);
+        await elementUpdated(el);
+        await nextFrame();
+  
+        expect(el.getAttribute('aria-valuetext')).to.be.equal('Time: 23:00');
+        expect(el.getAttribute('aria-valuenow')).to.be.equal(el.displayTime.toString());
+      });
+      it('Should increase minute value and update aria-valuetext when Arrow Up is pressed on minute segment', async () => {
+        await onTapstart(minutesSegment, el);
+        createKeyboardEvent(el, InputKey.ArrowUp);
+        await elementUpdated(el);
+        await nextFrame();
+  
+        expect(el.getAttribute('aria-valuetext')).to.be.equal('Time: 00:01');
+        expect(el.getAttribute('aria-valuenow')).to.be.equal(el.displayTime.toString());
+      });
+      it('Should decrease minute value and update aria-valuetext when Arrow Down is pressed on minute segment', async () => {
+        await onTapstart(minutesSegment, el);
+        createKeyboardEvent(el, InputKey.ArrowUp);
+        await elementUpdated(el);
+        await nextFrame();
+  
+        expect(el.getAttribute('aria-valuetext')).to.be.equal('Time: 00:01');
+        expect(el.getAttribute('aria-valuenow')).to.be.equal(el.displayTime.toString());
+      });
+      it('Switches segment to hours when Arrow Right is pressed', async () => {
+        await onTapstart(hoursSegment, el);
+        createKeyboardEvent(el, InputKey.ArrowRight);
+        await elementUpdated(el);
+        updateBtnSelector();
+
+        expect(el.activeSegment).to.be.equal('minutes');
+        expect(incrementBtnInHours).to.be.equal(null);
+        expect(decrementBtnInHours).to.be.equal(null);
```

Comment:
If you want to test that segment switches, only test for line 201. If you want to test that buttons should not show then do it in separate test case.